spyder/widgets/ipythonconsole/debugging.py

```diff
@@ -52,10 +54,14 @@ def refresh_from_pdb(self, pdb_state):
             self.sig_pdb_step.emit(fname, lineno)
 
         if 'namespace_view' in pdb_state:
-            self.sig_namespace_view.emit(pdb_state['namespace_view'])
+            if pdb_state['namespace_view']:
```

Comment:
I add this in order to pass the test `test_browse_history_dbg` since it was failing with a `malformed node or string` error, should change this for something like:

```python
            if pdb_state['namespace_view']:
                self.sig_namespace_view.emit(ast.literal_eval(
                        pdb_state['namespace_view']))
            else:
                self.sig_namespace_view.emit(
                        pdb_state['namespace_view'])
```